The latest commit fixes some issues where we used hardcoded /tmp paths in our tests, which obviously break with multiple tests using the same file. This was fixed by using `tempfile.mktemp` instead.

The last test failures look like this:
```
>       assert re.match(r"fd:\s+0x1 \((/dev/pts/\d+|pipe:\[\d+\])\)", line_fd)
E       AssertionError: assert None
E        +  where None = <function match at 0x7fbf446005e0>('fd:\\s+0x1 \\((/dev/pts/\\d+|pipe:\\[\\d+\\])\\)', 'fd:        0x1 (/tmp/parZ4YC4.par (deleted))')
E        +    where <function match at 0x7fbf446005e0> = re.match
```

--- a/setup-dev.sh
+++ b/setup-dev.sh
@@ -33,7 +33,8 @@ install_apt() {
         libc6-dev \
         curl \
         build-essential \
-        gdb
+        gdb \
+        parallel
 
     if [[ "$1" == "22.04" ]]; then
         sudo apt install shfmt

--- a/tests/gdb-tests/tests.sh
+++ b/tests/gdb-tests/tests.sh
@@ -70,8 +70,6 @@ fi
 
 TESTS_LIST=($(echo -E "$TESTS_COLLECT_OUTPUT" | grep -o "tests/.*::.*" | grep "${TEST_NAME_FILTER}"))
 
-declare -a FAILED_TESTS
-
 run_test() {
     test_case="$1"
 
@@ -89,21 +87,26 @@ run_test() {
 
     exit $?
 }
+JOBLOG_PATH="$(mktemp)"
+echo "Joblog: $JOBLOG_PATH"
 
 . $(which env_parallel.bash)
-env_parallel run_test ::: "${TESTS_LIST[@]}"
+env_parallel --joblog $JOBLOG_PATH run_test ::: "${TESTS_LIST[@]}"
 
-tests_failed=$?
-tests_passed_or_skipped=$((${#TESTS_LIST[@]} - $tests_failed))
+# The seventh column in the joblog is the exit value and the tenth is the test name
+FAILED_TESTS=($(awk '$7 == "1" { print $10 }' "${JOBLOG_PATH}"))
+
+num_tests_failed=${#FAILED_TESTS[@]}
+num_tests_passed_or_skipped=$((${#TESTS_LIST[@]} - $num_tests_failed))
 
 echo ""
 echo "*********************************"
 echo "********* TESTS SUMMARY *********"
 echo "*********************************"
-echo "Tests passed or skipped: ${tests_passed_or_skipped}"
-echo "Tests failed: ${tests_failed}"
+echo "Tests passed or skipped: ${num_tests_passed_or_skipped}"
+echo "Tests failed: ${num_tests_failed}"
 
-if [ "${tests_failed}" -ne 0 ]; then
+if [ "${num_tests_failed}" -ne 0 ]; then
     echo ""
     echo "Failing tests: ${FAILED_TESTS[@]}"
     exit 1

--- a/tests/gdb-tests/tests/test_context_commands.py
+++ b/tests/gdb-tests/tests/test_context_commands.py
@@ -32,8 +32,10 @@ def test_context_disasm_show_fd_filepath(start_binary):
 
     assert "call   read@plt" in line_call_read
 
+    # When running tests with GNU Parallel, sometimes the file name looks like
+    # this: /tmp/parZ4YC4.par (deleted)
     line_fd = line_fd.strip()
-    assert re.match(r"fd:\s+0x1 \((/dev/pts/\d+|pipe:\[\d+\])\)", line_fd)
+    assert re.match(r"fd:\s+0x1 \((/dev/pts/\d+|/tmp/par.+\.par \(deleted\)|pipe:\[\d+\])\)", line_fd)
 
     line_buf = line_buf.strip()
     assert re.match(r"buf:\s+0x[0-9a-f]+ ◂— 0x0", line_buf)

--- a/tests/gdb-tests/tests/test_context_commands.py
+++ b/tests/gdb-tests/tests/test_context_commands.py
@@ -32,10 +32,13 @@ def test_context_disasm_show_fd_filepath(start_binary):
 
     assert "call   read@plt" in line_call_read
 
-    # When running tests with GNU Parallel, sometimes the file name looks like
-    # this: /tmp/parZ4YC4.par (deleted)
+    # When running tests with GNU Parallel, sometimes the file name looks
+    # '/tmp/parZ4YC4.par', and occasionally '(deleted)' is present after the
+    # filename
     line_fd = line_fd.strip()
-    assert re.match(r"fd:\s+0x1 \((/dev/pts/\d+|/tmp/par.+\.par \(deleted\)|pipe:\[\d+\])\)", line_fd)
+    assert re.match(
+        r"fd:\s+0x1 \((/dev/pts/\d+|/tmp/par.+\.par(?: \(deleted\))?|pipe:\[\d+\])\)", line_fd
+    )
 
     line_buf = line_buf.strip()
     assert re.match(r"buf:\s+0x[0-9a-f]+ ◂— 0x0", line_buf)

